install/kubernetes/cilium/templates/cilium-secrets-namespace.yaml

```diff
@@ -16,3 +16,17 @@ kind: Namespace
 metadata:
   name: {{ .Values.gatewayAPI.secretsNamespace.name | quote }}
 {{- end}}
+
+# Only create the namespace if it's different from Ingress and Gateway API secret namespaces (if enabled).
+{{- if and .Values.envoyConfig.enabled .Values.envoyConfig.secretsNamespace.create .Values.envoyConfig.secretsNamespace.name
+  (and
+    (or (not (and .Values.ingressController.enabled .Values.ingressController.secretsNamespace.create .Values.ingressController.secretsNamespace.name))
+        (ne .Values.envoyConfig.secretsNamespace.name .Values.ingressController.secretsNamespace.name))
+    (or (not (and .Values.gatewayAPI.enabled .Values.gatewayAPI.secretsNamespace.create .Values.gatewayAPI.secretsNamespace.name))
+        (ne .Values.envoyConfig.secretsNamespace.name .Values.gatewayAPI.secretsNamespace.name))) }}
```

Comment:
I feel like I ran into a complexity limit in my brain evaluating this.

Reply:
like it or not, this is what peak helm performance looks like :runner:

Reply:
there's a better way to do this :-)

Make a dict of desired secret namespace names, something like

```

{{- $secretNamespaces := dict -}}
{{- range $cfg := tuple .Values.ingressController .Values.gatewayAPI .Values.envoyConfig -}}
{{- if and $cfg.enabled $cfg.secretsNamespace.create $cfg.secretsNamespace.name -}}
{{- $_ = set $secretNamespaces $cfg.secretsNamespace.name -}}
{{- end -}}
{{- end -}}

{{- range $name, _ := $secretNamespaces }}
---
apiVersion: v1
kind: Namespace
metadata:
  name: {{ $name | quote }}
{{- end}}
```

Reply:
I much prefer this approach to the current logic, which is not easy to read. This could use a comment along the lines of `{{/* Use a dict to create the set of unique secret namespaces */}}`